### PR TITLE
[Mode] Hyrule Warriors-style Link

### PR DESCRIPTION
**Pull request by OtherBlue**
Adds an option to place the Keaton and Fierce Deity masks similarly to how they are placed on Young Link in Hyrule Warriors once they are acquired in-game.

Branches off #537 for the "OnPlayerPostLimbDraw" hook

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2186970412.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2186970520.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2186977266.zip)

### CODE REVIEW

**Comment by garrettjoecox** [comment]
@OtherBlue nice work! Is your intent to have this merged upstream or just have a place to host it? If the latter you can follow the `modding.md` guide on Shipwright to create a PR into your own fork, in which it will still generate artifacts for people to download.

**Comment by PurpleHato** [comment]
Personally, I would love to see this making it's way to the main build
It's like a small pleasant addition, that is an existing modern design and doesn't recquire much edits and isn't invasive at all

Good job buddy!

**Comment by OtherBlue** [comment]
I'm thinking of opening the PR to the main build now

Wasn't the original intention but it's probably more convenient anyway

Just gotta figure out why clang-format is failing ig

File: mm/2s2h/Enhancements/Enhancements.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@ void InitEnhancements() {
 
     // Graphics
     RegisterDisableBlackBars();
+    RegisterHWStyledLink();
```
Comment (inspectredc) [inline mm/2s2h/Enhancements/Enhancements.cpp]: I think it'd be easier to understand what this is doing if we write it out in full
```suggestion
    RegisterHyruleWarriorsStyledLink();
```

File: mm/2s2h/Enhancements/Graphics/HWStyledLink.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+void UpdateHWStyledLink() {
+    GameInteractor::Instance->RegisterGameHookForID<GameInteractor::OnPlayerPostLimbDraw>(
+        PLAYER_LIMB_HEAD, [](Player* player, s32 limbIndex) {
+            if (CVarGetInteger("gModes.HWStyledLink", 0) && player->currentMask == PLAYER_MASK_NONE &&
+                player->transformation == PLAYER_FORM_HUMAN && INV_CONTENT(ITEM_MASK_KEATON) == ITEM_MASK_KEATON) {
+                OPEN_DISPS(gPlayState->state.gfxCtx);
+                Matrix_Push();
+                Matrix_RotateYS(14563, MTXMODE_APPLY);
+                Matrix_RotateZS(-4854, MTXMODE_APPLY);
```
Comment (inspectredc) [inline mm/2s2h/Enhancements/Graphics/HWStyledLink.cpp]: Matrix_RotateYS and Matrix_RotateZS use 'binangs' (binary angles). Would it be alright to write this as hex?

Alternatively, write it in degrees and use the DEG_TO_BINANG(angle) macro

**Review by inspectredc — COMMENTED** [review]
Looks Great! Just a couple things

File: mm/2s2h/Enhancements/Enhancements.cpp (hunk 1 above)
Comment (OtherBlue) [reply mm/2s2h/Enhancements/Enhancements.cpp]: Oh yeah, absolutely, just old habits still sticking around haha

File: mm/2s2h/Enhancements/Graphics/HWStyledLink.cpp (hunk 2 above)
Comment (OtherBlue) [reply mm/2s2h/Enhancements/Graphics/HWStyledLink.cpp]: tbh these first values were put in there by Proxy, I'll look into converting them tho

**Comment by OtherBlue** [comment]
Alrighty, final(?) changes made

File: mm/src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope_NES.c
Comment (mckinlee) [inline mm/src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope_NES.c:1]: This header is no longer needed.

File: mm/2s2h/Enhancements/Graphics/HyruleWarriorsStyledLink.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#include "HyruleWarriorsStyledLink.h"
+#include "libultraship/libultraship.h"
+#include "2s2h/Enhancements/GameInteractor/GameInteractor.h"
```
Comment (mckinlee) [inline mm/2s2h/Enhancements/Graphics/HyruleWarriorsStyledLink.cpp]: This has moved. Change to: #include "2s2h/GameInteractor/GameInteractor.h"

**Comment by OtherBlue** [comment]
Closed this by mistake, it's not going anywhere, my bad

**Review by garrettjoecox — APPROVED** [review]
Had been holding off on this one because I wasn't sure how I felt about it as an actual code addition, but it's well isolated so let's do it.

**Comment by julio-link002** [comment]
OtherBlue:HW-Styled-Link

**Pull request merged**